jenkins/bootstrap.py

```diff
@@ -330,10 +330,23 @@ def cat(self, path, generation):
         return self.call(cmd, output=True)
 
 
-    def upload_artifacts(self, path, artifacts):
+    def upload_artifacts(self, gsutil, path, artifacts):
         """Upload artifacts to the specified path."""
         # Upload artifacts
         if os.path.isdir(artifacts):
+            try:
+                # If remote path exists, it will create .../_artifacts subdir instead
+                gsutil.stat(path)
```

Comment:
There's technically a race here -- no guarantee that files were not uploaded here between the stat and cp call.

Maybe we should reach out to the gsutil team and see if there is a better solution here we can follow?